docs/node/workflow-external-dependencies.md

```diff
@@ -0,0 +1,57 @@
+# External Dependencies
+
+:::caution
+This is an advanced feature and should be used with care.
+:::
+
+Workflows in Temporal may be replayed from the beginning of their history when resumed.
+In order for Temporal to recreate the exact state Workflow code was in, the code is required to be [fully deterministic](/docs/node/determinism).
+To prevent breaking determinism, in the Node SDK, Workflow code runs in an isolated execution environment limited to functionality provided by the SDK.
+
+External Dependencies is an isolation breaking mechanism that allows injecting functions from the main NodeJS environment into a Workflow isolate.
+They are typically used in order to inject custom instrumentation (e.g. logger) functions into the isolate.
```

Comment:
interesting - isn't logging a common workflow, rather than an advanced one? dont we expect every production user to have to use this?

Reply:
Agreed, the doc is inaccurate.

Using `ASYNC_IGNORED` is completely safe and should commonly be used to propagate logs and metrics out of a Workflow.

I'll clarify that only external dependencies that return a value are dangerous to use.